x/examples/resolve/README.md

```diff
@@ -0,0 +1,91 @@
+# Domain Resolution Tool
+
+The `resolve` tool lets you resolve domain names with custom DNS resolvers and using configurable transports.
+
+Usage:
+
+```txt
+Usage: resolve [flags...] <domain>
+  -resolver string
+        The address of the recursive DNS resolver to use in host:port format. If the port is missing, it's assumed to be 53
+  -tcp
```

Comment:
You are right, -tcp wouldn't make sense for DoH/DoT. However, for now I'm just considering standard DNS.

I'm building up to something that can support multiple solutions. I'm trying to do it in a way that we can reuse the standard Resolver, so we can piggyback on the standard Dialer (which takes a Resolver). I'm not sure we can use DoH and reuse the standard Dialer, because Resolver is not that configurable, so we may need to reimplement a lot of the Dialer logic to use DoH.

For now, I've decided to punt on those complexities and stick to standard DNS, and we can revisit in a following step.

Reply:
I'll note, however, that this implementation supports DoT if we force TCP and add a TLS transport that allows us to set the alpn to `dot`, since the plaintext wire format is the same as DNS over TCP.